Update dockershim blog articles with content for after v1.24 is released (#33138)
kubernetes/website: content/en/blog/_posts/2022-02-17-updated-dockershim-faq.md

aliases: [ '/dockershim' ]
---

**This supersedes the original
[Dockershim Deprecation FAQ](/blog/2020/12/02/dockershim-faq/) article,
published in late 2020. The article includes updates from the v1.24
release of Kubernetes.**

This document goes over some frequently asked questions regarding the

> removal of _dockershim_ from Kubernetes. That removal was originally

Reviewer: nit

```suggestion
removal of _dockershim_ from Kubernetes. The removal was originally
```

Author: Not sure which is more idiomatic, but happy to revise it.

[announced](/blog/2020/12/08/kubernetes-1-20-release-announcement/)
as a part of the Kubernetes v1.20 release. The Kubernetes

> [v1.24 release](/releases/#release-v1-24) actually removed the dockershim

Reviewer: ```suggestion
[v1.24 release](/releases/#release-v1-24) completely removed the dockershim
```
or "fully removed" or "finalized the removal of"

Author: There was no partial removal; I'm worried that this implies that there was.

Reviewer: Yeah and this is fine as-is, thanks :)

from Kubernetes.

For more on on what that means, check out the blog post
[Don't Panic: Kubernetes and Docker](/blog/2020/12/02/dont-panic-kubernetes-and-docker/).

> You can read [check whether dockershim removal affects you](/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/)

Reviewer: Suggest changing the structure to "To determine the impact that the removal of dockershim would have for you or your organization, you can read [Check whether dockershim removal affects you](link).

to determine how much impact the removal of dockershim would have for you
or for your organization.

> As the Kubernetes 1.24 release became imminent, Kubernetes contributors worked hard to try to make this a smooth transition.

Reviewer: ```suggestion
In the months and days leading up to the Kubernetes 1.24 release, Kubernetes contributors worked hard to try to make this a smooth transition.
```

Additionally, features that were largely incompatible with the dockershim, such
as cgroups v2 and user namespaces are being implemented in these newer CRI

> runtimes. Removing support for the dockershim allows further development in those areas.

Reviewer: ```suggestion
runtimes. Removing dockershim allows further development in those areas.
```

Docker Engine, and to maintain that adapter even after the in-tree dockershim is removed
from Kubernetes. The replacement adapter is named [`cri-dockerd`](https://github.com/Mirantis/cri-dockerd).

> You can install `cri-dockerd` and use it to connect the kubelet to Docker Engine.

Author: Mentioned it. How's this?

> You can install `cri-dockerd` and use it to connect the kubelet to Docker Engine.

Reviewer: LGTM, thank you!
